.github/workflows/test-build-deploy.yml

```diff
@@ -338,17 +338,34 @@ jobs:
         uses: actions/download-artifact@v4
         with:
           name: Race-enabled Mimir
-      - name: Load race-enabled mimir into Docker
-        run: |
-          docker load -i ./mimir_race_image
-          export IMAGE_TAG_RACE=$(make image-tag-race)
-          docker run "grafana/mimir:$IMAGE_TAG_RACE" --version
       - name: Preload Images
         # We download docker images used by integration tests so that all images are available
         # locally and the download time doesn't account in the test execution time, which is subject
         # to a timeout
         run: go run ./tools/pre-pull-images | xargs -n1 -P4 docker pull
-      - name: Integration Tests
+      - name: Load race-enabled mimir-distroless into Docker
+        run: |
+          docker load -i ./mimir_race_image
+          export IMAGE_TAG_RACE=$(make image-tag-race)
+          docker run "grafana/mimir:$IMAGE_TAG_RACE-distroless" --version
+      - name: Integration Tests With Distroless Docker Image
+        run: |
+          export IMAGE_TAG_RACE=$(make image-tag-race)
+          export MIMIR_IMAGE="grafana/mimir:$IMAGE_TAG_RACE-distroless"
+          export IMAGE_TAG=$(make image-tag)
+          export MIMIRTOOL_IMAGE="grafana/mimirtool:$IMAGE_TAG"
+          export MIMIR_CHECKOUT_DIR="/go/src/github.com/grafana/mimir"
+          echo "Running integration tests with image: $MIMIR_IMAGE (Mimir), $MIMIRTOOL_IMAGE (Mimirtool)"
+          echo "Running integration tests (group ${{ matrix.test_group_id }} of ${{ matrix.test_group_total }}) with Go version: $(go version)"
+          ./.github/workflows/scripts/run-integration-tests-group.sh --index ${{ matrix.test_group_id }} --total ${{ matrix.test_group_total }}
+      - name: Load race-enabled mimir alpine into Docker
+        if: ${{ needs.checkDeploy.outputs.is_deploy == 'true' }}
+        run: |
+          docker load -i ./mimir_race_image
+          export IMAGE_TAG_RACE=$(make image-tag-race)
+          docker run "grafana/mimir:$IMAGE_TAG_RACE" --version
+      - name: Run also Integration Tests on alpine when deploying
```

Comment:
here we run always distroless image e2e tests, but we only run e2e test on alpine during release